### DISCUSSION TITLE
Docs: missing documentation for react compiler "compilationMode: annotation"

### DISCUSSION THREAD

**Discussion by Chenalejandro** [vercel/next.js#71994] [Help] [answered]
### What is the documentation issue?

How does the annotation works?

In the react website there is the `use no memo` directive (https://react.dev/learn/react-compiler#something-is-not-working-after-compilation), but I am not sure if the annotation mode is refering that.

From my testing, "use no memo" works by just setting this:
```
  experimental: {
    reactCompiler: true,
  },
```

The annotation mode in next.js makes me think that it is more an opt-in feature than the `use no memo`'s opt-out.

### Is there any context that might help us understand?

No

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/canary/app/api-reference/next-config-js/reactCompiler

**Comment by Chenalejandro** [comment, marked as answer]
Just found the annotation mode in the react 19 docs: https://19.react.dev/learn/react-compiler. So, to opt-in we should use the `use memo`  directive.

**Reply by liby** [reply]
I'd like to clarify my understanding of the `compilationMode: 'annotation'` mode:

When using `compilationMode: 'annotation'`, components are NOT optimized by default, and only components that explicitly use the `"use memo"` directive will be optimized by the compiler?

This would make it an opt-in approach, as opposed to the default mode (`reactCompiler: true`) which optimizes all components by default and allows opting out with `"use no memo"`.

Is this understanding correct?

**Reply by amarpatel** [reply]
I've confirmed your understanding as correct on my local env 👍
